File: Data Structures & Algorithms/house-robber-ii/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        def search(n):
            rob1, rob2 = 0, 0

            for num in n:
                temp = max(rob1+num, rob2)
                rob1 = rob2
                rob2 = temp
            return rob2

        if len(nums) == 1:
            return nums[0]

        return max(search(nums[:-1]), search(nums[1:]))

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const seen = new Set(nums);
        let longest = 0;

        for (let num of seen) {
        if (!seen.has(num - 1)) {
            let currLongest = 1;
            let curr = num;
            while (seen.has(curr + 1)) {
            currLongest += 1;
            curr += 1;
            }
            longest = Math.max(longest, currLongest);
        }
        }

        return longest;
    }
}
